example/lib/test_scenarios/sensor_tag_test_scenario.dart

```diff
@@ -1,97 +1,195 @@
+    log("Connecting to ${_peripheral.name}");
+    await _peripheral.connect();
+    log("Connected!");
+    return peripheral;
+  }
+
+  Future<Peripheral> _discovery(
+      Logger log, Logger logError, Peripheral peripheral) async {
+    await peripheral.discoverAllServicesAndCharacteristics();
+    List<Service> services = await peripheral.services();
+    log("PRINTING SERVICES for ${peripheral.name}");
+    services.forEach((service) => log("Found service ${service.uuid}"));
+    Service service = services.first;
+    log("PRINTING CHARACTERISTICS FOR SERVICE \n${service.uuid}");
+
+    List<Characteristic> characteristics = await service.characteristics();
+    characteristics.forEach((characteristic) {
+      log("${characteristic.uuid}");
     });
 
-    log("CONNECTING to ${peripheral.name}, ${peripheral.identifier}...");
-    await peripheral.connect();
-    log("CONNECTED to ${peripheral.name}, ${peripheral.identifier}!");
-    deviceConnectionAttempted = false;
-
-    await peripheral
-        .discoverAllServicesAndCharacteristics()
-        .then((_) => peripheral.services())
-        .then((services) {
-          log("PRINTING SERVICES for ${peripheral.name}");
-          services.forEach((service) => log("Found service ${service.uuid}"));
-          return services.first;
-        })
-        .then((service) async {
-          log("PRINTING CHARACTERISTICS FOR SERVICE \n${service.uuid}");
-          List<Characteristic> characteristics =
-              await service.characteristics();
-          characteristics.forEach((characteristic) {
-            log("${characteristic.uuid}");
-          });
-
-          log("PRINTING CHARACTERISTICS FROM \nPERIPHERAL for the same service");
-          return peripheral.characteristics(service.uuid);
-        })
-        .then((characteristics) => characteristics.forEach((characteristic) =>
-            log("Found characteristic \n ${characteristic.uuid}")))
-        .then((_) {
-          log("WAITING 10 SECOND BEFORE DISCONNECTING");
-          return Future.delayed(Duration(seconds: 10));
-        })
-        .then((_) {
-          log("DISCONNECTING...");
-          return peripheral.disconnectOrCancelConnection();
-        })
-        .then((_) {
-          log("Disconnected!");
-          log("WAITING 10 SECOND BEFORE DESTROYING CLIENT");
-          return Future.delayed(Duration(seconds: 10));
-        })
-        .then((_) {
-          log("DESTROYING client...");
-          return bleManager.destroyClient();
-        })
-        .then((_) => log("\BleClient destroyed after a delay"));
+    log("PRINTING CHARACTERISTICS FROM \nPERIPHERAL for the same service");
+    List<Characteristic> characteristicFromPeripheral =
+        await _peripheral.characteristics(service.uuid);
+    characteristicFromPeripheral.forEach((characteristic) =>
+        log("Found characteristic \n ${characteristic.uuid}"));
+    return peripheral;
+  }
+
+  Future<Peripheral> _rssi(
```

Comment:
could you change the method name to something more meaningful? a verb would be helpful